pkg/river/encoding/riverjson/riverjson.go

```diff
@@ -87,15 +87,6 @@ func encodeFieldAsStatements(prefix []string, field rivertags.Field, fieldValue
 		fullName := mergeStringSlice(prefix, field.Name)
 
 		switch {
-		case fieldValue.IsZero():
-			// It shouldn't be possible to have a required block which is unset, but
-			// we'll encode something anyway.
-			return []jsonStatement{jsonBlock{
-				Name: strings.Join(fullName, "."),
-				Type: "block",
-				Body: nil,
-			}}
```

Comment:
This block was in the wrong place after the logic for block slices was introduced. It should only run for struct blocks.